New field collected user on all AMQP payload originating from sabre

```
[twake-calendar-side-service] 12:05:56.945 [ERROR] c.l.c.a.ItipLocalDeliveryConsumer - Error consuming calendar:itip:localDelivery message
[twake-calendar-side-service] com.fasterxml.jackson.databind.exc.UnrecognizedPropertyException: Unrecognized field "connectedUser" (class com.linagora.calendar.amqp.ItipLocalDeliveryDTO), not marked as ignorable (8 known properties: "uid", "method", "sender", "message", "calendarId", "hasChange", "oldMessage", "recipients"])
[twake-calendar-side-service]  at [Source: REDACTED (`StreamReadFeature.INCLUDE_SOURCE_IN_LOCATION` disabled); line: 1, column: 1259] (through reference chain: com.linagora.calendar.amqp.ItipLocalDeliveryDTO["connectedUser"])
[twake-calendar-side-service] 	at com.fasterxml.jackson.databind.exc.UnrecognizedPropertyException.from(UnrecognizedPropertyException.java:61)
[twake-calendar-side-service] 	at com.fasterxml.jackson.databind.DeserializationContext.handleUnknownProperty(DeserializationContext.java:1180)
[twake-calendar-side-service] 	at com.fasterxml.jackson.databind.deser.std.StdDeserializer.handleUnknownProperty(StdDeserializer.java:2244)
[twake-calendar-side-service] 	at com.fasterxml.jackson.databind.deser.BeanDeserializerBase.handleUnknownProperty(BeanDeserializerBase.java:1823)
[twake-calendar-side-service] 	at com.fasterxml.jackson.databind.deser.BeanDeserializerBase.handleUnknownProperties(BeanDeserializerBase.java:1773)
[twake-calendar-side-service] 	at com.fasterxml.jackson.databind.deser.BeanDeserializer._deserializeUsingPropertyBased(BeanDeserializer.java:563)
[twake-calendar-side-service] 	at com.fasterxml.jackson.databind.deser.BeanDeserializerBase.deserializeFromObjectUsingNonDefault(BeanDeserializerBase.java:1499)
[twake-calendar-side-service] 	at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserializeFromObject(BeanDeserializer.java:340)
[twake-calendar-side-service] 	at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserialize(BeanDeserializer.java:177)
[twake-calendar-side-service] 	at com.fasterxml.jackson.databind.deser.DefaultDeserializationContext.readRootValue(DefaultDeserializationContext.java:342)
[twake-calendar-side-service] 	at com.fasterxml.jackson.databind.ObjectMapper._readMapAndClose(ObjectMapper.java:4971)
[twake-calendar-side-service] 	at com.fasterxml.jackson.databind.ObjectMapper.readValue(ObjectMapper.java:3948)
[twake-calendar-side-service] 	at com.linagora.calendar.amqp.ItipLocalDeliveryDTO.deserialize(ItipLocalDeliveryDTO.java:71)
[twake-calendar-side-service] 	... 17 common frames omitted
[twake-calendar-side-service] Wrapped by: java.lang.RuntimeException: Failed to deserialize ItipLocalDeliveryDTO: {"sender":"mailto:user_7669386f-ee85-472f-af9e-c20ff9beabc9@open-paas.org","method":"REQUEST","uid":"6e1e8ece-96ba-40d3-adcd-66bf6d9c9e8c","message":"BEGIN:VCALENDAR\r\nVERSION:2.0\r\nPRODID:-\/\/Sabre\/\/Sabre VObject 4.5.7\/\/EN\r\nCALSCALE:GREGORIAN\r\nMETHOD:REQUEST\r\nBEGIN:VTIMEZONE\r\nTZID:Asia\/Ho_Chi_Minh\r\nBEGIN:STANDARD\r\nTZOFFSETFROM:+0700\r\nTZOFFSETTO:+0700\r\nTZNAME:ICT\r\nDTSTART:19700101T000000\r\nEND:STANDARD\r\nEND:VTIMEZONE\r\nBEGIN:VEVENT\r\nUID:6e1e8ece-96ba-40d3-adcd-66bf6d9c9e8c\r\
[twake-calendar-side-service] 	at com.linagora.calendar.amqp.ItipLocalDeliveryDTO.deserialize(ItipLocalDeliveryDTO.java:74)
[twake-calendar-side-service] 	at com.linagora.calendar.amqp.ItipLocalDeliveryConsumer.lambda$consumeMessage$0(ItipLocalDeliveryConsumer.java:189)
[twake-calendar-side-service] 	at reactor.core.publisher.MonoCallable.call(MonoCallable.java:72)
[twake-calendar-side-service] 	at reactor.core.publisher.FluxFlatMap.trySubscribeScalarMap(FluxFlatMap.java:128)
[twake-calendar-side-service] 	at reactor.core.publisher.MonoFlatMap.subscribeOrReturn(MonoFlatMap.java:53)
[twake-calendar-side-service] 	at reactor.core.publisher.Mono.subscribe(Mono.java:4560)
[twake-calendar-side-service] 	at reactor.core.publisher.FluxFlatMap$FlatMapMain.onNext(FluxFlatMap.java:430)
[twake-calendar-side-service] 	at reactor.core.publisher.FluxUsing$UsingSubscriber.onNext(FluxUsing.java:202)
[twake-calendar-side-service] 	at reactor.core.publisher.FluxCreate$BufferAsyncSink.drain(FluxCreate.java:880)
[twake-calendar-side-service] 	at reactor.core.publisher.FluxCreate$BufferAsyncSink.next(FluxCreate.java:805)
[twake-calendar-side-service] 	at reactor.core.publisher.FluxCreate$SerializedFluxSink.next(FluxCreate.java:163)
[twake-calendar-side-service] 	at reactor.rabbitmq.Receiver.lambda$null$9(Receiver.java:216)
[twake-calendar-side-service] 	at com.rabbitmq.client.impl.recovery.AutorecoveringChannel$2.handleDelivery(AutorecoveringChannel.java:588)
[twake-calendar-side-service] 	at com.rabbitmq.client.impl.ConsumerDispatcher$5.run(ConsumerDispatcher.java:149)
[twake-calendar-side-service] 	at com.rabbitmq.client.impl.ConsumerWorkService$WorkPoolRunnable.run(ConsumerWorkService.java:111)
[twake-calendar-side-service] 	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
[twake-calendar-side-service] 	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
[twake-calendar-side-service] 	at java.base/java.lang.Thread.run(Unknown Source)
```

Ensure we can deserialze all AMQP messages if they have some extra fields we don't handle TBH.

--- a/calendar-amqp/src/main/java/com/linagora/calendar/amqp/EventAlarmConsumer.java
+++ b/calendar-amqp/src/main/java/com/linagora/calendar/amqp/EventAlarmConsumer.java
@@ -42,6 +42,7 @@ import org.apache.james.util.ReactorUtils;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
+import com.fasterxml.jackson.databind.DeserializationFeature;
 import com.fasterxml.jackson.databind.ObjectMapper;
 import com.fasterxml.jackson.datatype.jdk8.Jdk8Module;
 import com.github.fge.lambdas.Throwing;
@@ -61,7 +62,8 @@ import reactor.rabbitmq.Receiver;
 import reactor.rabbitmq.Sender;
 
 public class EventAlarmConsumer implements Closeable, Startable {
-    private static final ObjectMapper OBJECT_MAPPER = new ObjectMapper().registerModule(new Jdk8Module());
+    private static final ObjectMapper OBJECT_MAPPER = new ObjectMapper().registerModule(new Jdk8Module())
+        .configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);
     private static final Logger LOGGER = LoggerFactory.getLogger(EventAlarmConsumer.class);
     private static final boolean REQUEUE_ON_NACK = true;
 

--- a/calendar-amqp/src/main/java/com/linagora/calendar/amqp/EventCalendarConsumer.java
+++ b/calendar-amqp/src/main/java/com/linagora/calendar/amqp/EventCalendarConsumer.java
@@ -42,6 +42,7 @@ import org.apache.james.util.ReactorUtils;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
+import com.fasterxml.jackson.databind.DeserializationFeature;
 import com.fasterxml.jackson.databind.ObjectMapper;
 import com.fasterxml.jackson.datatype.jdk8.Jdk8Module;
 import com.github.fge.lambdas.Throwing;
@@ -60,7 +61,8 @@ import reactor.rabbitmq.Receiver;
 import reactor.rabbitmq.Sender;
 
 public class EventCalendarConsumer implements Closeable, Startable {
-    private static final ObjectMapper OBJECT_MAPPER = new ObjectMapper().registerModule(new Jdk8Module());
+    private static final ObjectMapper OBJECT_MAPPER = new ObjectMapper().registerModule(new Jdk8Module())
+        .configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);
     private static final Logger LOGGER = LoggerFactory.getLogger(EventCalendarConsumer.class);
     private static final boolean REQUEUE_ON_NACK = true;
 

--- a/calendar-amqp/src/main/java/com/linagora/calendar/amqp/EventEmailConsumer.java
+++ b/calendar-amqp/src/main/java/com/linagora/calendar/amqp/EventEmailConsumer.java
@@ -41,6 +41,7 @@ import org.apache.james.util.ReactorUtils;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
+import com.fasterxml.jackson.databind.DeserializationFeature;
 import com.fasterxml.jackson.databind.ObjectMapper;
 import com.fasterxml.jackson.datatype.jdk8.Jdk8Module;
 import com.google.inject.name.Named;
@@ -77,7 +78,8 @@ public class EventEmailConsumer implements Closeable, Startable {
     private static final boolean PUBLIC_AGENDA_EVENT = true;
     private static final Logger LOGGER = LoggerFactory.getLogger(EventEmailConsumer.class);
     private static final boolean REQUEUE_ON_NACK = true;
-    private static final ObjectMapper OBJECT_MAPPER = new ObjectMapper().registerModule(new Jdk8Module());
+    private static final ObjectMapper OBJECT_MAPPER = new ObjectMapper().registerModule(new Jdk8Module())
+        .configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);
 
     private final ReceiverProvider receiverProvider;
     private final EventMailHandler eventMailHandler;

--- a/calendar-amqp/src/main/java/com/linagora/calendar/amqp/EventResourceConsumer.java
+++ b/calendar-amqp/src/main/java/com/linagora/calendar/amqp/EventResourceConsumer.java
@@ -42,6 +42,7 @@ import org.apache.james.util.ReactorUtils;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
+import com.fasterxml.jackson.databind.DeserializationFeature;
 import com.fasterxml.jackson.databind.ObjectMapper;
 import com.fasterxml.jackson.datatype.jdk8.Jdk8Module;
 import com.github.fge.lambdas.Throwing;
@@ -60,7 +61,8 @@ import reactor.rabbitmq.Receiver;
 import reactor.rabbitmq.Sender;
 
 public class EventResourceConsumer implements Closeable, Startable {
-    private static final ObjectMapper OBJECT_MAPPER = new ObjectMapper().registerModule(new Jdk8Module());
+    private static final ObjectMapper OBJECT_MAPPER = new ObjectMapper().registerModule(new Jdk8Module())
+        .configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);
     private static final Logger LOGGER = LoggerFactory.getLogger(EventResourceConsumer.class);
     private static final boolean REQUEUE_ON_NACK = true;
 

--- a/calendar-amqp/src/main/java/com/linagora/calendar/amqp/ItipLocalDeliveryDTO.java
+++ b/calendar-amqp/src/main/java/com/linagora/calendar/amqp/ItipLocalDeliveryDTO.java
@@ -33,6 +33,7 @@ import com.fasterxml.jackson.annotation.JsonProperty;
 import com.fasterxml.jackson.core.JsonParser;
 import com.fasterxml.jackson.core.JsonToken;
 import com.fasterxml.jackson.databind.DeserializationContext;
+import com.fasterxml.jackson.databind.DeserializationFeature;
 import com.fasterxml.jackson.databind.JsonDeserializer;
 import com.fasterxml.jackson.databind.ObjectMapper;
 import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
@@ -56,7 +57,8 @@ public record ItipLocalDeliveryDTO(
     @JsonDeserialize(using = RecipientListDeserializer.class)
     @JsonProperty("recipients") List<String> recipients) {
 
-    private static final ObjectMapper OBJECT_MAPPER = new ObjectMapper().registerModule(new Jdk8Module());
+    private static final ObjectMapper OBJECT_MAPPER = new ObjectMapper().registerModule(new Jdk8Module())
+        .configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);
 
     public static byte[] serialize(ItipLocalDeliveryDTO dto) {
         try {

--- a/calendar-amqp/src/test/java/com/linagora/calendar/amqp/ItipLocalDeliveryDTOTest.java
+++ b/calendar-amqp/src/test/java/com/linagora/calendar/amqp/ItipLocalDeliveryDTOTest.java
@@ -103,6 +103,39 @@ class ItipLocalDeliveryDTOTest {
     }
 
     @Test
+    // Sabre started adding a "connectedUser" field to the payload: unknown fields must be ignored.
+    void deserializeShouldIgnoreUnknownFields() {
+        byte[] payload = """
+            {
+                "sender": "mailto:sender@example.com",
+                "method": "REQUEST",
+                "uid": "uid-123",
+                "calendarId": "calendar-123",
+                "message": "BEGIN:VCALENDAR",
+                "hasChange": true,
+                "connectedUser": "mailto:connected@example.com",
+                "recipients": [
+                    "mailto:alice@example.com"
+                ]
+            }
+            """.getBytes(StandardCharsets.UTF_8);
+
+        ItipLocalDeliveryDTO actual = ItipLocalDeliveryDTO.deserialize(payload);
+
+        ItipLocalDeliveryDTO expected = new ItipLocalDeliveryDTO(
+            "mailto:sender@example.com",
+            "REQUEST",
+            "uid-123",
+            "calendar-123",
+            "BEGIN:VCALENDAR",
+            Optional.empty(),
+            true,
+            List.of("mailto:alice@example.com"));
+
+        assertThat(actual).isEqualTo(expected);
+    }
+
+    @Test
     void strippedRecipientShouldRemoveMailtoPrefix() {
         ItipLocalDeliveryDTO dto = dtoWithRecipient("mailto:alice@example.com");
 

